.github/workflows/publish.yaml

```
@@ -8,19 +8,28 @@ jobs:
   build:
     name: Build distribution package
     runs-on: ubuntu-latest
-
     steps:
       - uses: actions/checkout@v4
         with:
           persist-credentials: false
+      - name: Fetch tags
+        run: git fetch --prune --unshallow --tags
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
           python-version: "3.x"
       - name: Install pypa/build
-        run: python -m pip install --user build
+        run: python -m pip install --user build twine
+      - name: Add '.devN' to version for non-tag builds
+        if: ${{ ! startsWith(github.ref, 'refs/tags/') }}
```

I find it more convenient to use `>-` for escaping stuff and making it less nested. This allows us to drop the mustache:
```suggestion
        if: >-
          !startsWith(github.ref, 'refs/tags/')
```

Though, there's a nicer way to check for the same thing nowadays, more elegant:

```suggestion
        if: github.ref_type != 'tag'
```

---

These are 2 good points.

Updated with your last suggestion.

(I also changed to `if: github.ref_type == 'tag'` in job `publish-to-pypi` for consistency)